packages/ai-openai/src/browser/openai-frontend-application-contribution.ts

```diff
@@ -38,72 +40,119 @@ export class OpenAiFrontendApplicationContribution implements FrontendApplicatio
+
+        const modelsToRemove = oldModels.filter(model => !newModels.some(newModel => newModel.id === model.id));
+        const modelsToAddOrUpdate = newModels.filter(newModel =>
+            !oldModels.some(model =>
+                model.id === newModel.id &&
+                model.model === newModel.model &&
+                model.url === newModel.url &&
+                model.apiKey === newModel.apiKey &&
+                model.defaultRequestSettings === newModel.defaultRequestSettings &&
+                model.enableStreaming === newModel.enableStreaming));
+
+        this.manager.removeLanguageModels(...modelsToRemove.map(model => model.id));
+        this.manager.createOrUpdateLanguageModels(...modelsToAddOrUpdate);
+        this.prevCustomModels = [...newCustomModels];
+    }
+
+    protected handleRequestSettingsChanges(newSettings: RequestSetting[]): void {
+        const models = this.preferenceService.get<string[]>(MODELS_PREF, []);
+        this.manager.createOrUpdateLanguageModels(...models.map(modelId => this.createOpenAIModelDescription(modelId, newSettings)));
+
+        const customModels = this.preferenceService.get<Partial<OpenAiModelDescription>[]>(CUSTOM_ENDPOINTS_PREF, []);
+        this.manager.createOrUpdateLanguageModels(...this.createCustomModelDescriptionsFromPreferences(customModels, newSettings));
+    }
+
+    protected createOpenAIModelDescription(modelId: string, requestSettings: RequestSetting[]): OpenAiModelDescription {
+        const id = `${OPENAI_ID}/${modelId}`;
+        const modelRequestSetting = this.getMatchingRequestSetting(modelId, OPENAI_ID, requestSettings);
+        return {
+            id: id,
+            model: modelId,
+            apiKey: true,
+            enableStreaming: !openAIModelsWithDisabledStreaming.includes(modelId),
+            defaultRequestSettings: modelRequestSetting?.requestSettings
+        };
+    }
+
+    protected createCustomModelDescriptionsFromPreferences(
+        preferences: Partial<OpenAiModelDescription>[],
+        requestSettings: RequestSetting[] = []
+    ): OpenAiModelDescription[] {
+        return preferences.reduce((acc, pref) => {
+            if (!pref.model || !pref.url || typeof pref.model !== 'string' || typeof pref.url !== 'string') {
+                return acc;
             }
-        ];
-    }, []);
+
+            const modelRequestSetting = this.getMatchingRequestSetting(pref.model, OPENAI_ID, requestSettings);
+
+            return [
+                ...acc,
+                {
+                    id: pref.id && typeof pref.id === 'string' ? pref.id : pref.model,
+                    model: pref.model,
+                    url: pref.url,
+                    apiKey: typeof pref.apiKey === 'string' || pref.apiKey === true ? pref.apiKey : undefined,
+                    enableStreaming: pref.enableStreaming ?? true,
+                    defaultRequestSettings: modelRequestSetting?.requestSettings ?? pref.defaultRequestSettings
```

Comment:
With this we now allow the defaultRequestSettings to also be defined within the custom open ai preferences. Is this intentional? If yes, then we also should add it to the preferences schema and documentation. Note that in the Llamafile preferences we do not support that and forcefully ignore such preferences in case the user did define them.

Reply:
I remove the ability